src/Workspaces/Core/Portable/Workspace/Solution/ProjectState.cs

```diff
@@ -296,8 +294,11 @@ internal sealed class ProjectAnalyzerConfigOptionsProvider : AnalyzerConfigOptio
             public ProjectAnalyzerConfigOptionsProvider(ProjectState projectState)
                 => _projectState = projectState;
 
+            private AnalyzerConfigOptionsCache GetCache()
+                => _projectState._lazyAnalyzerConfigOptions.GetValue(CancellationToken.None);
+
             public override AnalyzerConfigOptions GlobalOptions
-                => GetOptionsForSourcePath(string.Empty);
```

Comment:
Seems like a bug. It doesn't seem like the compiler returns global options when empty path is passed to `GetOptionsForSourcePath`.